cognite/neat/rules/_analysis/_information_rules.py

```diff
@@ -0,0 +1,341 @@
+
+        Args:
+            rules: Instance of InformationRules holding the data model
+            consider_inheritance: Whether to consider inheritance or not. Defaults False
+
+        Returns:
+            Set of classes that are connected to other classes
+        """
+        class_linkage = self.class_linkage(consider_inheritance)
+        return set(class_linkage.source_class.values).union(set(class_linkage.target_class.values))
+
+    def defined_classes(self, consider_inheritance: bool = False) -> set[ClassEntity]:
+        """Returns classes that have properties defined for them in the data model.
+
+        Args:
+            rules: Instance of InformationRules holding the data model
+            consider_inheritance: Whether to consider inheritance or not. Defaults False
+
+        Returns:
+            Set of classes that have been defined in the data model
+        """
+        class_property_pairs = self.classes_with_properties(consider_inheritance)
+        properties = list(itertools.chain.from_iterable(class_property_pairs.values()))
+
+        return {property.class_ for property in properties}
+
+    def disconnected_classes(self, consider_inheritance: bool = False) -> set[ClassEntity]:
+        """Return a set of classes that are disconnected (i.e. isolated) from other classes.
+
+        Args:
+            rules: Instance of InformationRules holding the data model
+            consider_inheritance: Whether to consider inheritance or not. Defaults False
+
+        Returns:
+            Set of classes that are disconnected from other classes
+        """
+        return self.defined_classes(consider_inheritance) - self.connected_classes(consider_inheritance)
+
+    def symmetrically_connected_classes(
+        self, consider_inheritance: bool = False
+    ) -> set[tuple[ClassEntity, ClassEntity]]:
+        """Returns a set of pairs of symmetrically linked classes.
+
+        Args:
+            rules: Instance of InformationRules holding the data model
+            consider_inheritance: Whether to consider inheritance or not. Defaults False
+
+        Returns:
+            Set of pairs of symmetrically linked classes
+        """
+
+        # TODO: Find better name for this method
+        sym_pairs: set[tuple[ClassEntity, ClassEntity]] = set()
+
+        class_linkage = self.class_linkage(consider_inheritance)
+        if class_linkage.empty:
+            return sym_pairs
+
+        for _, row in class_linkage.iterrows():
+            source = row.source_class
+            target = row.target_class
+            target_targets = class_linkage[class_linkage.source_class == target].target_class.values
+            if source in target_targets and (source, target) not in sym_pairs:
+                sym_pairs.add((source, target))
+        return sym_pairs
+
+    def as_property_dict(self) -> dict[str, list[InformationProperty]]:
+        """This is used to capture all definitions of a property in the data model."""
```

Comment:
Is it possible to have multiple definition of the same property? Shouldn't that be an error?